packages/vite/src/node/plugins/preAlias.ts

```diff
@@ -14,7 +14,9 @@ export function preAliasPlugin(): Plugin {
       server = _server
     },
     resolveId(id, importer, options) {
-      if (!options?.ssr && bareImportRE.test(id)) {
+      const optimizeDeps = server._optimizeDepsMetadata
+      const hasPreBundled = optimizeDeps && optimizeDeps.optimized[id]
+      if (!options?.ssr && (bareImportRE.test(id) || hasPreBundled)) {
```

Comment:
I think we can inline the prebundle check into the if condition too: `hasPreBundled` => `server._optimizeDepsMetadata?.optimized[id]`.

Reply:
Ok，it's a better way.

Reply:
Thanks for the quick fix! I was hoping that the expression would be inside the if condition though so we don't unnecessarily run the hasPreBundled expression. The if condition should then be formatted and span multiple lines by prettier.

Reply:
You are right.Emmm.. I think using `hasPreBundled` expression can make it understood easier.

Reply:
I will optimize it. And can i leave a comment to explain the motivation of the condition?

Reply:
Yeah a comment would be great :+1: